src/foolscap/test/test_banana.py

```diff
@@ -1787,13 +1945,21 @@ def _testIdentity_1(self, z):
         self.assertIdentical(z[0][0], z)
 
     def testUnicode(self):
-        x = [unicode('blah')]
+        x = [six.ensure_text('blah')]
         d = self.loop(x)
         d.addCallback(self._testUnicode_1, x)
         return d
     def _testUnicode_1(self, y, x):
-        self.assertEquals(x, y)
-        self.assertEquals(type(x[0]), type(y[0]))
+
+        if type(x) in (str, bytes):
+            x = six.ensure_text(x)
+            y = six.ensure_text(y)
+        elif type(x) in (list, tuple):
+            x = list(map(six.ensure_text, x))
+            y = list(map(six.ensure_text, y))
+            
```

Comment:
should use list-comprehensions, not `map` in new code

Reply:
Changed. Thank you.